### DISCUSSION TITLE
Unable to edit Pub settings

### DISCUSSION THREAD

**Discussion by lemixtape** [knowledgefutures/pubpub#1763] [Support]
Hi,

whenever I try to edit a Pub and click the _Save changes_ button, I get no error message and everything seems fine. When I try to leave the settings page however, I get a pop-up that asks me if I want to leave the page while content is not saved. If I leave the page and come back, the edits I made have not been applied.

Does anyone know why the changes are not saved? Do you need a paid plans to edit Pubs?

**Comment by gabestein** [comment]
I'm having trouble reproducing this. Can you let me know the browser and operating system you're using, and the exact steps we can use to replicate the issue?

**Comment by lemixtape** [comment]
Here is a video that replicates the issue. I have tried to do this in multiple browsers, on 2 different computers, and I always get the same result. 

I could only host it on my Dropbox: https://www.dropbox.com/s/xzsey4gv223c6p5/Edit%20settings.mp4?dl=0

**Reply by gabestein** [reply]
Thanks for this, and I'm sorry about the issue. The problem appears to be related to the browser URL not updating after you change the slug name. I don't have this problem on Chrome 97.

So we can track down the issue, can you please let me know the exact browser versions you tried on?

In the meantime, you should be able to make settings changes by changing the slug and then browsing to the Pub settings again at the new slug.

**Comment by lemixtape** [comment]
I am using Windows 10. I tried on Chrome 97 and Firefox 96. Both are not
working unfortunately.

I am not sure I understand what you mean by your solution. Would it be
possible for you to send me a bit more detailed instructions?

Thank you for your patience.

**Reply by gabestein** [reply]
Thanks. I've been able to figure this out — it actually wasn't related to browser, as I thought, but an error condition we're not reporting well. Long story short, because of some legacy decisions, Pub slugs are unique across all of PubPub, not just within communities. We have some ideas for solving that, but at the moment, someone else has `/pub/cv`. When you try to change it to that, the server is returning an error, as it should. But PubPub isn't surfacing to you properly, which it absolutely should do.

The workaround is to choose another slug for now and make sure that the url updates when you change it.

In the short term, we'll add a warning when this happens.

In the longer term, we have a plan to make it so slug collisions like this aren't a problem.

**Reply by augustfly** [reply]
hi @gabestein -- are pub slugs still unique across all of PubPub? Asking bc our publication rate is going to go up this year and need to maybe rethink our slug model. Thanks! Gus

**Reply by gabestein** [reply]
hi Gus — yup, they're still unique.

**Reply by augustfly** [reply]
Thanks; we've published some bad slugs and are trying to manually fix them.

**Comment by lemixtape** [comment]
Great, glad you could figure it out, and that it exposed a bug! One less
bug until PubPub is perfect now.

Thank you for the great support, I really appreciate it.

David
